### ISSUE TITLE
[STM32RET6]: Can‘t connect to target. After using mdk to flash.

### ISSUE THREAD

**Issue by cnjhb** [stlink-org/stlink#1275]
- [x] I made serious effort to avoid creating duplicate or nearly similar issue
- [x] Programmer/board type: [V2] 
- [x] Operating system an version: [Linux] 
- [x] **stlink tools version** and/or git commit hash: [1.7.0] 
- [x] stlink commandline tool name: [st-flash]
- [x] Target chip (and board, if applicable): [stm32ret6] 

Commandline output:

```
[joseph@hikari stm32ret6usart]$ st-flash --connect-under-reset --format ihex write build/stm32ret6usart.hex
st-flash 1.7.0
2022-10-28T16:22:59 WARN common.c: NRST is not connected
2022-10-28T16:22:59 ERROR common.c: Soft reset failed: timeout
2022-10-28T16:22:59 ERROR common.c: Can not connect to target. Please use 'connect under reset' and try again
Failed to connect to target
```

description:

`Only after using mdk to flash`
`  It can be repaired after flashing with stm32cubeide. `

**Comment by Ehsan2754** [comment]
Due to previous experience of mine here are my suggestions.
* double check connections
* double check boot pin configuration
* make sure you have already enabled SWD on your controller before flashing. try to erase current firmware on ur device somehow and program again with SWD enabled.

**Comment by Ehsan2754** [comment]
Due to previous experience of mine here are my suggestions.
* double check connections
* double check boot pin configuration
* make sure you have already enabled SWD on your controller before flashing. try to erase current firmware on ur device somehow and program again with SWD enabled.

**Comment by Nightwalker-87** [comment]
There is nothing to add to the feedback from @Ehsan2754 from my side.
This may likely be an hardware related issue.
Unfortunately we can't investigate this any further due to the limited amount of information available.
Please open a new ticket if the problem still persists; then with a full debug output (`st-flash --debug write`).

**Comment by Nightwalker-87** [comment]
As appropriate advice has been given, we're closing this ticket now.
